File: reference_functions/ex1.js
class Point {
  constructor(a, b) {
    this.a = a;
    this.b = b;
  }
}

function attribute(point, x, z) {
  point.a = x;
  point.b = z;
}

function main() {
  const p = new Point(10, 2);

  //   const p = { a: 10, b: 5 };

  attribute(p, 1, 2);

  console.log(`p -> a = ${p.a}`);
  console.log(`p -> b = ${p.b}`);
}

main();

File: reference_functions/ex1.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int a;
    int b;
} Point;

void attribute(Point * p, int x, int z);

int main () {
    Point *p = NULL;

    p = (Point*)malloc(sizeof(Point));

    if(p == NULL) {
        printf("Error to allocate memory");
        exit(1);
    }

    p->a = 10;
    p->b = 20;

    attribute(p, 1, 2);

    printf("p -> a = %i \n", p->a );
    printf("p -> b = %i \n", p->b );

    free(p);

    return 0;
}

void attribute(Point *p, int x, int z) {
    p->a = x;
    p->b = z;
}
